presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestMinMaxByAggregation.java

```diff
@@ -297,4 +299,238 @@ public void testMaxShortDecimalDecimal()
                 createShortDecimalsBlock("1.1", "2.2", "3.3", "4.4"),
                 createShortDecimalsBlock("1.2", "1.0", "2.0", "1.5"));
     }
+
+    @Test
+    public void testMinBooleanVarchar()
+    {
+        InternalAggregationFunction function = METADATA.getFunctionRegistry().getAggregateFunctionImplementation(new Signature("min_by", AGGREGATE, parseTypeSignature(StandardTypes.VARCHAR), parseTypeSignature(StandardTypes.VARCHAR), parseTypeSignature(StandardTypes.BOOLEAN)));
+        assertAggregation(
+                function,
+                "b",
+                createStringsBlock("a", "b", "c"),
+                createBooleansBlock(true, false, true));
+    }
+
+    @Test
+    public void testMaxBooleanVarchar()
+    {
+        InternalAggregationFunction function = METADATA.getFunctionRegistry().getAggregateFunctionImplementation(new Signature("max_by", AGGREGATE, parseTypeSignature(StandardTypes.VARCHAR), parseTypeSignature(StandardTypes.VARCHAR), parseTypeSignature(StandardTypes.BOOLEAN)));
+        assertAggregation(
+                function,
+                "c",
+                createStringsBlock("a", "b", "c"),
+                createBooleansBlock(false, false, true));
+    }
+
+    @Test
+    public void testMinBooleanLongArray()
+    {
+        InternalAggregationFunction function = METADATA.getFunctionRegistry().getAggregateFunctionImplementation(new Signature("min_by", AGGREGATE, parseTypeSignature("array(bigint)"), parseTypeSignature("array(bigint)"), parseTypeSignature(StandardTypes.BOOLEAN)));
+        assertAggregation(
+                function,
+                null,
+                createArrayBigintBlock(asList(asList(3L, 4L), null, null)),
+                createBooleansBlock(true, false, true));
+    }
+
+    @Test
+    public void testMaxBooleanLongArray()
+    {
+        InternalAggregationFunction function = METADATA.getFunctionRegistry().getAggregateFunctionImplementation(new Signature("max_by", AGGREGATE, parseTypeSignature("array(bigint)"), parseTypeSignature("array(bigint)"), parseTypeSignature(StandardTypes.BOOLEAN)));
+        assertAggregation(
+                function,
+                asList(2L, 2L),
+                createArrayBigintBlock(asList(asList(3L, 4L), null, asList(2L, 2L))),
+                createBooleansBlock(false, false, true));
+    }
+
+    @Test
+    public void testMinLongVarchar()
```

Comment:
Can we add a test for `MinIntVarchar()` as well? I suspect the current code will fail?